huntsman/observatory.py

```diff
@@ -232,202 +233,67 @@ def autofocus_cameras(self, *args, **kwargs):
 
         return result
 
-    def take_evening_flats(self,
-                           alt=None,
-                           az=None,
-                           min_counts=5000,
-                           max_counts=15000,
-                           bias=1000,
-                           max_exptime=60.,
-                           camera_list=None,
-                           target_adu_percentage=0.5,
-                           max_num_exposures=10,
-                           *args, **kwargs
-                           ):
-        """Take flat fields
+    def take_flat_fields(self, camera_names=None, alt=None, az=None, **kwargs):
+        """
+        Take flat fields for each camera in each filter, respecting filter priority.
 
         Args:
-            alt (float, optional): Altitude for flats
-            az (float, optional): Azimuth for flats
-            min_counts (int, optional): Minimum ADU count
-            max_counts (int, optional): Maximum ADU count
-            bias (int, optional): Default bias for the cameras
-            max_exptime (float, optional): Maximum exposure time before stopping
-            camera_list (list, optional): List of cameras to use for flat-fielding
+            camera_names (list, optional): List of camera names to take flats with.
+            alt (float, optional): Altitude for flats.
+            az (float, optional): Azimuth for flats.
+            min_counts (int, optional): Minimum ADU count.
+            max_counts (int, optional): Maximum ADU count.
+            bias (int, optional): Default bias for the cameras.
+            max_exptime (float, optional): Maximum exposure time before stopping.
```

Comment:
I actually implemented `min_exptime` as a parameter already but didn't put it in the docstring. Will update.

Reply:
Needs a bit more to actually make it work than just the parameter though. The next exposure time calculation is rounding (or rather truncating) to integers so that will prevent exposure times less than 1 second too. Would need to think about how to go below 1 while still limiting the number of possible exposure times to a reasonable number. I was thinking a geometric progression would be a way to go, e.g. rounding to the nearest power of 2 so we have 1, 2, 4, 8 seconds, but also 1/2, 1/4, 1/8.

That's for another PR though, if we find we really need to go below 1 second.